### PR TITLE
Use ShellEvaluator in java shell

### PR DESCRIPTION
**Pull request by kornilova203**
There are couple of commits that were cherry-picked from my [java-shell-0.0.3](https://github.com/kornilova203/mongosh/tree/java-shell-0.0.3) branch and adapted to current code in master.
Java shell from this branch is used in DataGrip consoles.
Now I want to pull changes from this branch to master.

Most of the changes are in Java codebase, there is no need to review them, all java tests pass. But there is a change in mongosh core:
Shell evaluator fetches cursors and returns arrays with result. In DataGrip we want to have control over cursor and fetch it ourselves. This way we are able to start rendering results without waiting for whole fetch to be done.

See:
1. [`asShellResult` implementation in `AggregationCursor`](https://github.com/mongodb-js/mongosh/compare/master...kornilova203:use-shellevaluator-in-java-shell?expand=1#diff-685046018c1a23ff5d025a0eb4f1cb5aR47)
2. [`asShellResult` implementation in `Cursor`](https://github.com/mongodb-js/mongosh/compare/master...kornilova203:use-shellevaluator-in-java-shell?expand=1#diff-bfce9978e5907f6b28eb5446165997ebR33)

### CODE REVIEW

File: packages/shell-api/src/aggregation-cursor.ts
Comment (mcasimir) [inline packages/shell-api/src/aggregation-cursor.ts:50]: @aherlihy this should be fine right?

Comment (aherlihy) [reply packages/shell-api/src/aggregation-cursor.ts:50]: Yup, unless you want this to happen for all the classes (then we can move it into the decorators). If it works then fine by me

File: packages/shell-api/src/aggregation-cursor.ts
Comment (aherlihy) [inline packages/shell-api/src/aggregation-cursor.ts:16]:
```suggestion
import { asShellResult } from './enums';
```

Comment (aherlihy) [reply packages/shell-api/src/aggregation-cursor.ts:16]: We need single quotes for our linter to pass

**Review by aherlihy — CHANGES_REQUESTED** [review]
Looks great! I didn't look too closely at the java changes but I trust that it's all right :)

There are some small changes required to get our tests passing (we never bothered to set `ReplPlatform` for our tests since we never used it until now). I made 2 suggestions on this PR so that we can get our linter passing, but apparently I can't make suggestions on files you didn't modify on github so I made a separate PR to this repo with the changes that are required for the tests: https://github.com/mongodb-js/mongosh/pull/357. I am waiting for the CI to pass and I will merge it.

Once that PR is merged, the suggestions on this PR are accepted and this PR is rebased with master, our CI tests should hopefully pass and we can merge this.

**Comment by kornilova203** [comment]
I rebased the branch and executed `npm test` locally. All test pass, so I'll merge this branch

**Pull request merged**
